packages/builder/builder-shared/package.json

```diff
@@ -67,6 +68,7 @@
     }
   },
   "dependencies": {
+    "@modern-js/babel-preset-app": "workspace:*",
```

Comment:
Most Rspack projects do not need babel, if we declare the "@modern-js/babel-preset-app" dependency in `builder-shared` by default, the install size will be bigger.

Reply:
Maybe in long time, the app-tools would default dependent the `webpack-provider`, It is mean we already dependent the `@modern-js/babel-preset-app` by default. 

That's a problem exactly, but it 's not a problem now.
We should resolve this when we switch the bundler engine to rspack by default

Reply:
emm. Maybe we need think carefully about this problem.

Reply:
Updated it.`@modern-js/babel-preset-app` would be a dev dependency in `builder-shared`.

but the `babel-loader` still in `builder-shared`, There are two main reasons follow: 
- `babel-loader` only 242 kb size.
- The preBuild dependency does not load well dynamically